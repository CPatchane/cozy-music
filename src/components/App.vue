<template>
  <div class='app-wrapper o-layout--2panes'>
    <aside class='o-sidebar'>
      <nav>
        <ul class='c-nav'>
          <li class='c-nav-item'>
            <router-link class='c-nav-link' to='/viewhello1'>
              <icon class='c-nav-icon' v-bind:icon='navIcon' />
              {{ t('Nav.hello_nav_1') }}
            </router-link>
          </li>
          <li class='c-nav-item'>
            <router-link class='c-nav-link' to='/viewhello2'>
              <icon class='c-nav-icon' v-bind:icon='navIcon' />
              {{ t('Nav.hello_nav_2') }}
            </router-link>
          </li>
          <li class='c-nav-item'>
            <router-link class='c-nav-link' to='/viewhello3'>
              <icon class='c-nav-icon' v-bind:icon='navIcon' />
              {{ t('Nav.hello_nav_3') }}
            </router-link>
          </li>
        </ul>
      </nav>
    </aside>
    <main class='app-content'>
      <router-view class="view"></router-view>
    </main>
  </div>
</template>

<script>
import Vue from 'vue'
import VueRouter from 'vue-router'
import navIcon from '../assets/icons/icon-bullet-point.svg'

import icon from './Icon'

import Hello1 from './HelloViews/Hello1.vue'
import Hello2 from './HelloViews/Hello2.vue'
import Hello3 from './HelloViews/Hello3.vue'

// This installs <router-view> and <router-link>,
// and injects $router and $route to all router-enabled child components
Vue.use(VueRouter)

const router = new VueRouter({
  mode: 'hash',
  base: '/',
  linkActiveClass: 'is-active',
  routes: [
    { path: '/viewhello1', component: Hello1 },
    { path: '/viewhello2', component: Hello2 },
    { path: '/viewhello3', component: Hello3 },
    { path: '/', redirect: '/viewhello1'},
    // default, always at the end
    { path: '*', redirect: '/viewhello1'}
  ]
})

export default {
  router,
  components: { icon },
  data () {
    return {
      navIcon
    }
  },
}
</script>
