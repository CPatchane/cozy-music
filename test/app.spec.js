'use strict'

/* eslint-env jest */

import Vue from 'vue'
import { shallow } from '@vue/test-utils'

import App from '../src/components/App'
import I18nTestMixin from './lib/I18n'
Vue.mixin(I18nTestMixin)

describe('App component only', () => {
  it('should be mounted correctly', () => {
    const component = shallow(App)
    expect(component.html()).toMatchSnapshot()
  })
})
