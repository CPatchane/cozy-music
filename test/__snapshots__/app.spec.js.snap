// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component only should be mounted correctly 1`] = `
<div class="app-wrapper o-layout--2panes">
  <aside class="o-sidebar">
    <nav>
      <ul class="c-nav">
        <li class="c-nav-item">
          <a href="#/viewhello1" class="c-nav-link router-link-exact-active is-active">
            <!---->
            First
          </a>
        </li>
        <li class="c-nav-item">
          <a href="#/viewhello2" class="c-nav-link">
            <!---->
            Second
          </a>
        </li>
        <li class="c-nav-item">
          <a href="#/viewhello3" class="c-nav-link">
            <!---->
            Third
          </a>
        </li>
      </ul>
    </nav>
  </aside>
  <main class="app-content">
    <p style="margin-left: 2em;" class="view">Just... Hello world! This is View 1.</p>
  </main>
</div>
`;
